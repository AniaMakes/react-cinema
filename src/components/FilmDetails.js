import React from 'react';

class FilmDetails extends React.Component {
	constructor(props) {
		super(props)
	}

	render () {
		return (
			<div className='film-details'
				style={this.props.detailsShown ? {display: 'block'} : {display: 'none'}}
			>
				<p className='film-runtime'> {this.props.data.Runtime} </p>
				<p className='film-genre'> {this.props.data.Genre} </p>
				<p className='film-plot'> {this.props.data.Plot}</p>
			</div>
		)
	}
}



export default FilmDetails;