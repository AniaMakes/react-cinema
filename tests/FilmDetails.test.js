import React from 'react';
import FilmDetails from '../src/components/FilmDetails';
import renderer from 'react-test-renderer';

describe('Film Details Item', () => {
	it('films details item with \'display-details\' classname is rendered', () => {

		const data1 = {
			Runtime: '106m',
			Genre: 'comedy',
			Plot: 'plot here'
		}
		
		const mock = renderer.create(<FilmDetails 
			detailsShown={true}
			data={ data1}
			/>).toJSON();

		expect(mock).toMatchSnapshot();

	})
})

describe('Film Details Item', () => {
	it('films details item with \'hide-details\' classname is rendered', () => {

		const data1 = {
			Runtime: '106m',
			Genre: 'comedy',
			Plot: 'plot here'
		}
		
		const mock = renderer.create(<FilmDetails 
			detailsShown={false}
			data={ data1}
			/>).toJSON();

		expect(mock).toMatchSnapshot();

	})
})

