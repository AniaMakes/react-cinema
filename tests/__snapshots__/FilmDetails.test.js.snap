// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film Details Item films details item with 'display-details' classname is rendered 1`] = `
<div
  className="display-details"
>
  <p
    className="film-runtime"
  >
     
    <b>
      Runtime:
    </b>
     
    106m
     
  </p>
  <p
    className="film-genre"
  >
     
    <b>
       Genre:
    </b>
     
    comedy
     
  </p>
  <p
    className="film-plot"
  >
     
    plot here
  </p>
</div>
`;

exports[`Film Details Item films details item with 'hide-details' classname is rendered 1`] = `
<div
  className="hide-details"
>
  <p
    className="film-runtime"
  >
     
    <b>
      Runtime:
    </b>
     
    106m
     
  </p>
  <p
    className="film-genre"
  >
     
    <b>
       Genre:
    </b>
     
    comedy
     
  </p>
  <p
    className="film-plot"
  >
     
    plot here
  </p>
</div>
`;
